src/firecracker/src/api_server/parsed_request.rs

```diff
@@ -89,6 +90,8 @@ impl TryFrom<&Request> for ParsedRequest {
             (Method::Put, "boot-source", Some(body)) => parse_put_boot_source(body),
             (Method::Put, "cpu-config", Some(body)) => parse_put_cpu_config(body),
             (Method::Put, "drives", Some(body)) => parse_put_drive(body, path_tokens.next()),
+            #[cfg(target_arch = "x86_64")]
+            (Method::Put, "hotplug", Some(body)) => parse_put_hotplug(body),
```

Comment:
I also think that this could make sense. It would require some change, as there already exists PATCH for machine config, and therefore hotplug code would need to be changed to fit into what already exists. My main initial concern is that, currently, the logic of hotplugging is part of the `Vmm` struct, whereas the `machine-config` PATCH is handled in the VmResources struct. The current hotplugging logic uses both the seccomp filters and vcpu_exit_evt attributes of the Vmm, and so if it were to be moved into VmResources there would need to be some refactoring in order to achieve the same effect. I haven't scoped out how much change would be required to do this, but I feel as if this is something I can revisit later, as it likely won't change the performance of hotplugging, which is the main interest.

Reply:
This change would also clean up the updating of the machine config after hotplugging. Right now it's a bit a messy, as it's essentially another API call's worth of work without actually using the API. Not the neatest bit of code.

Reply:
I would disagree. The `PATCH` for `machine-config` seems worse, because not all fields can be patched (we cannot turn off smt or change cpu template on the fly). Separate API endpoint feels more correct to me. I don't want to have same story as with block API where we need to decide which fields can be present, which cannot and so on.

Reply:
My motivation is "cleaniness" of the API rather than easiness of the implementation, but I don't see the problem with adding a check on the fields that are being updated. However, if you think this is a problem we could even have a different body for the request, which will include only the fields that can be updated. We do that with various other devices, e.g. [block](https://github.com/firecracker-microvm/firecracker/blob/f5906eeb47f71f1ba412b24786d6762fc06dcbf4/src/firecracker/swagger/firecracker.yaml#L1141), [net](https://github.com/firecracker-microvm/firecracker/blob/f5906eeb47f71f1ba412b24786d6762fc06dcbf4/src/firecracker/swagger/firecracker.yaml#L1156).
 
In any case, as I said, I don't see this as a blocker for this PR.